Add pulsar 2.7.1 release note (#9834)
apache/pulsar: site2/website/release-notes.md

## Apache Pulsar Release Notes

> ### 2.7.1 &mdash; 2020-11-25 <a id=“2.7.1”></a>

Reviewer: 2020-11-25 should be changed to the current release date.

> ### 2.7.1 &mdash; 2020-11-25 <a id=“2.7.1”></a>

Author: ```suggestion
### 2.7.1 &mdash; 2021-03-18 <a id=“2.7.1”></a>
```

#### Broker

> - Fix topic ownership is not checked when get topic policy [#9781](https://github.com/apache/pulsar/pull/9781)

Reviewer 2: ```suggestion
- Fix topic ownership is not checked when getting topic policy [#9781](https://github.com/apache/pulsar/pull/9781)
```

> - Fix Cannot consume using older subscriptions if max subscription limit is reached [#9758](https://github.com/apache/pulsar/pull/9758)

Reviewer: I was wondering whether we need to refine the PR description, you can consult @Anonymitaet on it.

> - Fix Cannot consume using older subscriptions if max subscription limit is reached [#9758](https://github.com/apache/pulsar/pull/9758)

Reviewer 2: ```suggestion
- Fix the issue of consumers cannot be created for older subscriptions if the limit of `maxSubscriptionsPerTopic` is reached [#9758](https://github.com/apache/pulsar/pull/9758)
```

- Fix marking individual deletes as dirty [#9732](https://github.com/apache/pulsar/pull/9732)

> - Don't add broker-address header if response has already been committed [#9744](https://github.com/apache/pulsar/pull/9744)

Reviewer: If it's a fix, maybe we can describe it in another way, such as
Add broker address header name and check the commit response before adding a header.
You can consult Yu and make a decision whether to refine it.

- Keep topic-level policies commands consistent with that for namespace level [#9215](https://github.com/apache/pulsar/pull/9215)
- Fix handle topic loading failure due to broken schema ledger [#9212](https://github.com/apache/pulsar/pull/9212)
- Fix issue with topic compaction when compaction ledger is empty [#9206](https://github.com/apache/pulsar/pull/9206)
- Fix incoming message size issue that introduced in #9113 [#9182](https://github.com/apache/pulsar/pull/9182)
- Disallow parsing of token with none signature in authenticateToken [#9172](https://github.com/apache/pulsar/pull/9172)
- Fix locking for ConsumerImpl when creating deadLetterProducer [#9166](https://github.com/apache/pulsar/pull/9166)
- Fix maxProducersPerTopic cannot be disabled at the namespace level [#9157](https://github.com/apache/pulsar/pull/9157)
- Fix wrong default value [#9149](https://github.com/apache/pulsar/pull/9149)
- Fix the closed ledger did not delete after expired [#9136](https://github.com/apache/pulsar/pull/9136)
- Additional error checks in TwoPhasesCompactor [#9133](https://github.com/apache/pulsar/pull/9133)

> - Fix master broker while subscribe to non-persistent partitioned topic without topic auto-creation [#9107](https://github.com/apache/pulsar/pull/9107)

Reviewer 2: ```suggestion
- Fix master broker while subscribing to non-persistent partitioned topics without topic auto-creation [#9107](https://github.com/apache/pulsar/pull/9107)
```

- Support chained authentication with same auth method name [#9094](https://github.com/apache/pulsar/pull/9094)
- Broker expires messages one at a time after topic unload [#9083](https://github.com/apache/pulsar/pull/9083)
- Add refresh authentication command in broker [#9064](https://github.com/apache/pulsar/pull/9064)
- Add updateRates method for kop to collect publish rate [#9094](https://github.com/apache/pulsar/pull/9049)
- Fix DelayedDelivery at the broker level has a default value [#9030](https://github.com/apache/pulsar/pull/9030)
- Getting the stats of a non-persistent topic that has been cleaned causes it to re-appear [#9029](https://github.com/apache/pulsar/pull/9029)

> - Add raw prometheus metrics provider [#9021](https://github.com/apache/pulsar/pull/9021)

Reviewer 2: ```suggestion
- Add raw Prometheus metrics provider [#9021](https://github.com/apache/pulsar/pull/9021)
```

- Improve error handling when broker doesn't trust client certificates [#8998](https://github.com/apache/pulsar/pull/8998)

> - remove duplicated broker prometheus metrics type [8995](https://github.com/apache/pulsar/pull/8995)

Reviewer 2: ```suggestion
- Remove duplicated broker Prometheus metrics type [8995](https://github.com/apache/pulsar/pull/8995)
```

- Support configure max subscriptions per topic on the topic level policy [#8948](https://github.com/apache/pulsar/pull/8948)
- Fix subscription dispatch rate does not work after the topic unload without dispatch rate limit [#8947](https://github.com/apache/pulsar/pull/8947)
- Avro custom schema not working in consumer [#8939](https://github.com/apache/pulsar/pull/8939)
- Expose non-contiguous deleted messages ranges stats [#8936](https://github.com/apache/pulsar/pull/8936)
- Intercept beforeSendMessage calls [#8932](https://github.com/apache/pulsar/pull/8932)
- Monitor if a cursor moves its mark-delete position [#8930](https://github.com/apache/pulsar/pull/8930)
- Capture stats with precise backlog [#8928](https://github.com/apache/pulsar/pull/8928)
- Support configure max subscriptions per topic on the namespace level policy [#8924](https://github.com/apache/pulsar/pull/8924)
- Export Prometheus metric for messageTTL [#8871](https://github.com/apache/pulsar/pull/8871)
- Add pulsar-perf new feature: one subscription has more than one consumer [#8837](https://github.com/apache/pulsar/pull/8837)

> - Execute removing non-persistent subscription of topic from different thread to avoid deadlock when removing inactive subscriptions [#8820](https://github.com/apache/pulsar/pull/8820)

Reviewer 2: ```suggestion
- Execute removing non-persistent subscription of a topic from a different thread to avoid deadlock when removing inactive subscriptions [#8820](https://github.com/apache/pulsar/pull/8820)
```

#### Proxy

- Fix Proxy Config bindAddress does not working for servicePort [#9068](https://github.com/apache/pulsar/pull/9068)
- Return correct authz and auth errors from proxy to client [#9055](https://github.com/apache/pulsar/pull/9055)
- Fix the metadata setup compatibility issue [#8959](https://github.com/apache/pulsar/pull/8959)
- Support HAProxy proxy protocol for broker and proxy [#8686](https://github.com/apache/pulsar/pull/8686)

#### Pulsar Perf

- Dump JVM information [#9769](https://github.com/apache/pulsar/pull/9769)
- pulsar-perf uses DefaultCryptoKeyReader for E2E encryption  [#9668](https://github.com/apache/pulsar/pull/9668)
- Add --batch-index-ack for the pulsar-perf [#9521](https://github.com/apache/pulsar/pull/9521)

#### Transaction

- Fix deleteTransactionMarker memory leak [#9752](https://github.com/apache/pulsar/pull/9752)
- Fix transaction messages order error and deduplication error [#9024](https://github.com/apache/pulsar/pull/9024)
- Fix transaction log replay not handle right [#8723](https://github.com/apache/pulsar/pull/8723)

#### Pulsar Admin

- Validate offload param [#9737](https://github.com/apache/pulsar/pull/9737)

> - Inform user when expire message request is not executed. [#9561](https://github.com/apache/pulsar/pull/9561)

Reviewer 2: ```suggestion
- Inform user when expiring message request is not executed. [#9561](https://github.com/apache/pulsar/pull/9561)
```

- Fix get-message-by-id throwing NPE when message is null [#9537](https://github.com/apache/pulsar/pull/9537)
- Expire message by position [#9519](https://github.com/apache/pulsar/pull/9519)
- Add subscription backlog size info for topicstats [#9302](https://github.com/apache/pulsar/pull/9302)
- Expose schema ledger in `topic stats-internal` [#9284](https://github.com/apache/pulsar/pull/9284)

> - Fix potential http get hangs in the Pulsar Admin [#9203](https://github.com/apache/pulsar/pull/9203)

Reviewer 2: ```suggestion
- Fix potential HTTP get hangs in the Pulsar Admin [#9203](https://github.com/apache/pulsar/pull/9203)
```

- Fix admin-api-brokers list failed [#9191](https://github.com/apache/pulsar/pull/9191)
- Fix force delete namespace did not delete all topics of the namespace [#8806](https://github.com/apache/pulsar/pull/8806)
- Change method `getWebServiceUrl` into async [#8746](https://github.com/apache/pulsar/pull/8746)

> - Cannot get lastMessageId for an empty topic due to message retention [#8725](https://github.com/apache/pulsar/pull/8725)

Reviewer: Is the description an issue or a fix?
